### PR TITLE
Bundle shared libs in jar, so it becomes easier to install

### PR DESCRIPTION
**Pull request by atoulme**
This PR allows to search for libevmc in paths, and will install locally a libevmc as a temporary file if needed if the .so file is not found in the path.
This PR also resolves the test example-vm.so file to the Java tests so they don't depend on the user dir.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 /.idea
 Cargo.lock
 /target
+bindings/java/java/src/main/resources
+bindings/java/java/src/test/resources
```
Comment (axic) [inline .gitignore:7]: These should go into `bindings/java/.gitignore`.

File: bindings/java/c/host.c [hunk 2]
```diff
@@ -40,31 +40,28 @@ static bool account_exists_fn(struct evmc_host_context* context, const evmc_addr
     const char java_method_signature[] = "(I[B)I";
     assert(context != NULL);
     JNIEnv* jenv = attach();
-    if (jenv != NULL)
```
Comment (axic) [inline bindings/java/c/host.c:43]: Why did you remove these?

Comment (atoulme) [reply bindings/java/c/host.c:43]: cmake fails to compile with this. Also, the behavior is surprising - if JNI fails to attach, we should fail, not avoid executing the operation.

File: .gitignore (hunk 1 above)
Comment (atoulme) [reply .gitignore]: done

**Comment by atoulme** [comment]
Closing this in favor of #531

File: bindings/java/c/host.c (hunk 2 above)
Comment (axic) [reply bindings/java/c/host.c:43]: Can you show an error log?

Comment (atoulme) [reply bindings/java/c/host.c:43]:
```/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:370:9: error: variable 'result' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    if (jenv != NULL)
        ^~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:402:12: note: uninitialized use occurs here
    return result;
           ^~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:370:5: note: remove the 'if' if its condition is always true
    if (jenv != NULL)
    ^~~~~~~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:365:5: note: variable 'result' is declared here
    struct evmc_result result;
    ^
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:412:9: error: variable 'result' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    if (jenv != NULL)
        ^~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:439:12: note: uninitialized use occurs here
    return result;
           ^~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:412:5: note: remove the 'if' if its condition is always true
    if (jenv != NULL)
    ^~~~~~~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:407:5: note: variable 'result' is declared here
    struct evmc_tx_context result;
    ^
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:450:9: error: variable 'result' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
    if (jenv != NULL)
        ^~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:481:12: note: uninitialized use occurs here
    return result;
           ^~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:450:5: note: remove the 'if' if its condition is always true
    if (jenv != NULL)
    ^~~~~~~~~~~~~~~~~
/Users/antoinetoulme/w/evmc/bindings/java/c/host.c:445:5: note: variable 'result' is declared here```

Comment (axic) [reply bindings/java/c/host.c:43]: I think the preference would be to individually fix these issues, in separate PRs. I agree this extra conditional isn't the nicest, and probably we have multiple options:
1) Move the declarations around so the compiler won't complain
2) Instead of the conditional, throw a Java exception?
3) Instead of conditional, add an assertion (does an abort).
